src/main/java/domain/Cars.java

```diff
@@ -0,0 +1,49 @@
+package domain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Cars {
+
+    private List<Car> cars;
+
+    private Cars(List<Car> cars) {
+        this.cars = cars;
+    }
+
+    public static Cars create(String[] carNames, NumberGenerator numberGenerator) {
+        List<Car> newCars = new ArrayList<Car>();
+        for (String carName : carNames) {
+            newCars.add(new Car(carName, numberGenerator));
+        }
+        return new Cars(newCars);
+    }
+
+    public void move() {
+        for (Car car : cars) {
+            car.move();
+        }
+    }
+
+    public int getMaxDistance() {
+        int maxDistance = 0;
+        for (Car car : cars) {
+            maxDistance = Math.max(maxDistance, car.getDistance());
+        }
+        return maxDistance;
+    }
+
+    public Cars findCarsInPosition(int position) {
+        List<Car> carsInPosition = new ArrayList<>();
+        return new Cars(
+                cars.stream()
+                        .filter(car -> car.isInPosition(position))
+                        .collect(Collectors.toList())
+        );
```

Comment:
Stream을 잘 활용 하셨네요!

하나의 라인에서 여러 동작이 이루어져서 depth가 깊어졌네요.
괄호 안에서 여러 연산이 이루어지면 가독성이 많이 떨어져요. 글은 왼쪽에서부터 읽지만, 코드 흐름(연산)은 가장 안쪽의 괄호에서 이루어지잖아요!
```suggestion
        List<Car> carsInPosition = cars.stream()
                .filter(car -> car.isInPosition(position))
                .collect(Collectors.toList());
        return new Cars(carsInPosition);
```
이렇게 분리해서 적용하면 좋아보여요! 이름을 주어서 stream 연산이 어떤 역할을 수행 하는지도 표현 할 수 있겠죠?